src/wallet/wallet2.cpp

```diff
@@ -10473,14 +10479,14 @@ size_t wallet2::import_multisig(std::vector<cryptonote::blobdata> blobs)
   return n_outputs;
 }
 //----------------------------------------------------------------------------------------------------
-std::string wallet2::encrypt(const std::string &plaintext, const crypto::secret_key &skey, bool authenticated) const
+std::string wallet2::encrypt(const char *plaintext, size_t len, const crypto::secret_key &skey, bool authenticated) const
```

Comment:
This is an `epee::span<char>`. I find it preferrable to group the arguments into an object for clarity. Also the call for both `std::string` and `epee::wipeable_string` is just : `epee::to_span(plaintext)`. The overloads could actually be dropped, since its easy to call `encrypt` with that object.

Reply:
Not sure I like that. pointer and size_t is canonical everywhere. I'm sure you're more used to span and it looks simpler to you, it just doesn't to me. I could certainly add an overload with span too though. I'll think a bit more about this and try it.

Reply:
After trying this, it looks like your span class needs a writeable buffer.
While I could add a ctor from const data, it seems dangerous.
What do you suggest ?

Reply:
Add an overload for `char* wipeable_string::data()`, and an overload for `epee::to_mut_byte_span`. I actually added the latter in a patch I am working on. I don't see substantial difference to allowing a mutable pointer to the internal data, if a non-mutable pointer is going to be provided (because the bigger issue is calling code copying the secret data).

The canonical ptr+length arguments ... maybe ? LMDB used `MDB_val`. Several other libraries use similar constructs.

Reply:
Can you PR that span code alone (I asume it stands alone) ? Then I can use it directly.